node/rpc/src/rpc_util.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use log::{debug, error};
+use serde::de::DeserializeOwned;
+
+use crate::data_types::JsonRpcServerState;
+
+pub fn get_error_obj(code: i64, message: String) -> jsonrpc_v2::Error {
+    jsonrpc_v2::Error::Full {
+        code,
+        message,
+        data: None,
+    }
+}
+
+pub fn get_error_res(code: i64, message: String) -> jsonrpc_v2::ResponseObject {
+    jsonrpc_v2::ResponseObject::Error {
+        jsonrpc: jsonrpc_v2::V2,
+        error: get_error_obj(code, message),
+        id: jsonrpc_v2::Id::Null,
+    }
+}
+
+pub fn get_error_str(code: i64, message: String) -> String {
+    match serde_json::to_string(&get_error_res(code, message)) {
+        Ok(err_str) => err_str,
+        Err(err) => format!("Failed to serialize error data. Error was: {}", err),
+    }
+}
+
+pub const RPC_METHOD_CHAIN_HEAD_SUB: &str = "Filecoin.ChainHeadSubscription";
+pub const RPC_METHOD_CHAIN_NOTIFY: &str = "Filecoin.ChainNotify";
+pub const RPC_METHOD_CHAIN_NOTIFY_RESPONSE: &str = "xrpc.ch.val";
+
+const STREAMING_METHODS: [&str; 1] = [RPC_METHOD_CHAIN_NOTIFY];
```

Comment:
I think you missed the other two as well :p

Reply:
There are others?

Reply:
Filecoin.ChainHeadSubscription and xrpc.ch.val?

Reply:
Technically only ChainNotify is streaming, the others are traditional request-response. I can make a note of that if you like?

Reply:
Youre right. Lets make a note of that. I was more making sure that you cant call those other two methods from the HTTP endpoint.

Reply:
I've removed one, I don't think it's necessary to disable HTTP calls to the other.

Reply:
I think we should prevent `Filecoin.ChainHeadSubscription` from being called from HTTP still.
Its not like they can poll for the subscriptions anyways.

Reply:
Fair enough. It also does provide more balance to the code so it doesn't look like an oversight. Done.